Docs: improve docs narrative around "ENSNode Plugins", "ENSDb Writers", and "ENSDb Readers". (#2227)
namehash/ensnode: docs/ensnode.io/src/content/docs/docs/integrate/integration-options/ensdb-readers.mdx

---
title: ENSDb Readers (Custom APIs)

> description: How to build custom APIs on top of ENSDb, using the ENSDb Reader specification.

Reviewer: ```suggestion
description: How to build custom APIs and under services on top of ENSDb, using the ENSDb Reader standard.
```

---

import { LinkCard } from "@astrojs/starlight/components";

> 

Reviewer: @tk-o We need to introduce what an ENSDb Reader is up here. Please make some nice content for this.

> ## Standards-compliant ENSDb Readers

Reviewer: ```suggestion
## Current ENSDb Readers
```

There are multiple [ENSDb Reader](/docs/services/ensdb/concepts/glossary#ensdb-reader) apps created (and planned) by the NameHash team. Find out more about each of them below.

> 
> :::note[Build your own ENSDb Reader]
> Continuing on the ideas shared in [ENSDb integration options page](/docs/integrate/integration-options/ensdb), you can build your own ENSDb Reader app and define a custom data model and capabilities that are required for your use cases. Feel free to reach out to the NameHash team on [Telegram](https://t.me/ensnode) if you want to build your own ENSDb Reader and want any help or guidance!
> :::

Reviewer: ```suggestion
```

### ENSApi

> ENSApi is a reference implementation of an [ENSDb Reader](/docs/services/ensdb/concepts/glossary#ensdb-reader) that includes standards-compliant web services:

Reviewer: ```suggestion
ENSApi is a reference implementation of an [ENSDb Reader](/docs/services/ensdb/concepts/glossary#ensdb-reader) that includes the following APIs:
```

> - A standards-compliant [ENS Omnigraph GraphQL API](/docs/integrate/integration-options/omnigraph-graphql-api) on top of [ENSNode Plugins](/docs/integrate/integration-options/ensnode-plugins) such as `unigraph` and `protocol-acceleration`.

Reviewer: ```suggestion
- The new [ENS Omnigraph GraphQL API](/docs/integrate/integration-options/omnigraph-graphql-api) that's built on top of [ENSNode Plugins](/docs/integrate/integration-options/ensnode-plugins) such as `unigraph` and `protocol-acceleration`.
```

> - A standards-compliant [ENS Subgraph GraphQL API](/docs/integrate/ens-subgraph) on top of [ENSNode Plugins](/docs/integrate/integration-options/ensnode-plugins) such as `subgraph`, `basenames`, `lineanames`, and `threedns`.

Reviewer: ```suggestion
- A backwards-compatible [ENS Subgraph GraphQL API](/docs/integrate/ens-subgraph) that's built on top of [ENSNode Plugins](/docs/integrate/integration-options/ensnode-plugins) such as `subgraph`, `basenames`, `lineanames`, and `threedns`.
```

> - An API for accessing the metadata stored in ENSDb, including [metadata](/docs/services/ensdb/concepts/glossary#indexing-metadata-context) about the indexing status from the [ENSDb Writer](/docs/services/ensdb/concepts/glossary#ensdb-writer) and [metadata](/docs/services/ensdb/concepts/glossary#indexing-metadata-context) about the overall [ENSNode stack of services](/docs/services) active in the ENSNode instance.

Reviewer: ```suggestion
- APIs for various queries related to the the metadata stored in ENSDb, such as the indexing status from the [ENSDb Writer](/docs/services/ensdb/concepts/glossary#ensdb-writer) and [metadata](/docs/services/ensdb/concepts/glossary#indexing-metadata-context) about the overall [ENSNode stack of services](/docs/services) active in the ENSNode instance.
```

<LinkCard
  title="Learn more about ENSApi"
  description="A reference implementation of ENSDb Reader, built by the NameHash team."
  href="/docs/services/ensapi"
/>

> ### Other ENSDb Readers

Reviewer: ```suggestion
### New ENSDb Readers Under Development
```

> <LinkCard

Reviewer: ```suggestion
Multiple new [ENSDb Reader](/docs/services/ensdb/concepts/glossary#ensdb-reader) services are planned by the NameHash team. Find out more about each of them below.

<LinkCard
```

  title="Learn more about ensdb-cli"
  description="A command-line interface for interacting with ENSDb."
  href="/docs/integrate/integration-options/ensdb-cli"
/>

<LinkCard
  title="Learn more about ENSEngine"

>   description="A push-based web service allowing to track ENS data changes in real-time."

Reviewer: ```suggestion
  description="A push-based web service allowing you to track ENS data changes as they occur."
```

  href="/docs/integrate/integration-options/ensengine"
/>

<LinkCard
  title="Learn more about ENSAnalytics (coming soon)"
  description="A web service that provides analytics and insights on ENS data."
  href="#"

> />

Reviewer: ```suggestion
/>

:::note[Build your own ENSDb Reader]
Creating your own ENSDb Reader service enables you to build your own completely custom APIs and data pipeline services. See related inspiration on the [ENSDb integration options page](/docs/integrate/integration-options/ensdb). Use any programming language or framework you wish! If you'd like any help or guidance feel welcome to reach out to the NameHash team on [Telegram](https://t.me/ensnode).
:::
```

> />

Reviewer: @tk-o Please action this important feedback from Copilot.
